src/DotNetWorker/Hosting/FunctionsApplicationBuilder.cs

```diff
@@ -0,0 +1,134 @@
+
+    /// <inheritdoc/>
+    public ConfigurationManager Configuration => _hostApplicationBuilder.Configuration;
+
+    /// <inheritdoc/>
+    public IHostEnvironment Environment { get; private set; }
+
+    /// <inheritdoc/>
+    public ILoggingBuilder Logging => _hostApplicationBuilder.Logging;
+
+    /// <inheritdoc/>
+    public IMetricsBuilder Metrics => _hostApplicationBuilder.Metrics;
+
+    /// <inheritdoc />
+    public IServiceCollection Services => _hostApplicationBuilder.Services;
+
+    /// <summary>
+    /// Only available to AspNetCore extension (and maybe not even there in the future)
+    /// </summary>
+    internal IHostBuilder HostBuilder => _bootstrapHostBuilder;
+
+    /// <summary>
+    /// Placeholder
+    /// </summary>
+    /// <returns>placeholder</returns>
+    public IHost Build()
+    {
+        if (_hasBuilt)
+        {
+            throw new InvalidOperationException("The application has already been built.");
+        }
+
+        // we skipped this; make sure we add it at the end
+        this.UseDefaultWorkerMiddleware();
+
+        _hasBuilt = true;
+
+        return _hostApplicationBuilder.Build();
+    }
+
+    /// <inheritdoc/>
+    public void ConfigureContainer<TContainerBuilder>(IServiceProviderFactory<TContainerBuilder> factory, Action<TContainerBuilder>? configure = null) where TContainerBuilder : notnull
+    {
+        _hostApplicationBuilder.ConfigureContainer(factory, configure);
+    }
+
+    private IFunctionsWorkerApplicationBuilder InitializeHosting(BootstrapHostBuilder bootstrapHostBuilder)
+    {
+        bootstrapHostBuilder.ApplyServiceProviderFactory();
+        bootstrapHostBuilder.RunAndClearRegisteredCallbacks();
+
+        IFunctionsWorkerApplicationBuilder functionsWorkerApplicationBuilder = null!;
+
+        foreach (var descriptor in Services)
+        {
+            if (descriptor.ServiceType == typeof(IFunctionsWorkerApplicationBuilder))
+            {
+                functionsWorkerApplicationBuilder = (IFunctionsWorkerApplicationBuilder)descriptor.ImplementationInstance!;
+                break;
+            }
+        }
+
+        // Grab the HostBuilderContext from the property bag to use in the ConfigureWebHostBuilder. Then
+        // grab the IWebHostEnvironment from the webHostContext. This also matches the instance in the IServiceCollection.
+        //var hostContext = (HostBuilderContext)bootstrapHostBuilder.Properties[typeof(HostBuilderContext)];
+        Environment = bootstrapHostBuilder.Context.HostingEnvironment;
+
+        //Host = new ConfigureHostBuilder(bootstrapHostBuilder.Context, Configuration, Services);
+        //WebHost = new ConfigureWebHostBuilder(webHostContext, Configuration, Services);
+
+        return functionsWorkerApplicationBuilder;
+    }
+
+    /// <inheritdoc/>
+    public IFunctionsWorkerApplicationBuilder Use(Func<FunctionExecutionDelegate, FunctionExecutionDelegate> middleware)
```

Comment:
Based on the comment above, to more closely align with the ASP.NET Core model maybe we want to make these middleware be something that's registered on the application and not the builder? Unless there's a particular reason they need to be finalized before `Build` is called?

Reply:
Yeah, we started discussing this also. Today we generate the pipeline via DI but I don't think there's anything strictly requiring this. We discussed briefly what'd it take to mirror AspNet here and move this to the application. We'll follow up on this separately.